### PR TITLE
Fix out of range check

### PR DESCRIPTION
**Pull request by yasahi-hpc**
This PR aims at fixing a bug in `is_out_of_range_value_included` function.
Previous implementation would give a wrong result if `axes` variable is not sorted.

For example, it returns `false` (now returns `true`) in the previous implementation.
```C++
std::array<int> v2 = {0, 4, 1};
KokkosFFT::Impl::is_out_of_range_value_included(
      v2, static_cast<IntType>(2));
```

Following changes are made
- [x] Fix `is_out_of_range_value_included` function with `std::max_element`
- [x] Gives a `runtime_error` if a negative number is included
- [x] Add an unit_test for these cases

### CODE REVIEW

**Review by tpadioleau — DISMISSED** [review]
LGTM

File: common/src/KokkosFFT_utils.hpp
Comment (yasahi-hpc) [reply common/src/KokkosFFT_utils.hpp:78]: That is a good idea. Thank you for the comments

**Comment by yasahi-hpc** [comment]
I have updated accordingly.
@tpadioleau Can I have your second review please?

**Review by tpadioleau — APPROVED** [review]
Good to me

**Pull request merged**
